scipy/conftest.py

```diff
@@ -39,6 +39,30 @@ def pytest_runtest_setup(item):
     if mark is not None and np.intp(0).itemsize < 8:
         pytest.xfail('Fails on our 32-bit test platform(s): %s' % (mark.args[0],))
 
+    try:
+        from threadpoolctl import threadpool_limits
+        HAS_THREADPOOLCTL = True
+    except ImportError:
+        # Optional dependency only
+        HAS_THREADPOOLCTL = False
+
+    if HAS_THREADPOOLCTL:
+        # Set the number of openmp threads based on the number of workers
+        # xdist is using to prevent oversubscription. Simplified version of what
+        # sklearn does (it can rely on threadpoolctl and its builtin OpenMP helper
+        # functions)
+        try:
+            xdist_worker_count = int(os.environ['PYTEST_XDIST_WORKER_COUNT'])
+        except KeyError:
+            # raises when pytest-xdist is not installed
+            return
+
+        if not os.getenv('OMP_NUM_THREADS'):
+            max_openmp_threads = os.cpu_count() // 2  # use nr of physical cores
+            threads_per_worker = max(max_openmp_threads // xdist_worker_count, 1)
+            threadpool_limits(threads_per_worker, user_api='blas')
```

Comment:
This is broken for old versions of OpenBLAS, see joblib/threadpoolctl#88.

Reply:
@rgommers "old versions of OpenBLAS" includes versions used in some of the CI. You need another `try-except`.

Reply:
Thanks, adding.